floodlight/io/statsperform.py

```diff
@@ -155,55 +128,100 @@ def _read_open_event_csv_single_line(
     return event, team, segment
 
 
-def create_links_from_open_tracking_data_csv(
-    filepath_tracking: Union[str, Path]
-) -> Dict[str, Dict[int, int]]:
-    """Parses the entire open StatsPerform event data csv file for unique jIDs
-    (jerseynumbers) and creates a dictionary linking jIDs to xIDs in ascending order.
+def read_teamsheets_from_open_data_csv(
+    filepath_csv: Union[str, Path]
+) -> Dict[str, Teamsheet]:
+    """Parses the entire open StatsPerform tracking data CSV file for unique jIDs
+    (jerseynumbers) and creates teamsheets for both teams.
 
     Parameters
     ----------
-    filepath_tracking: str or pathlib.Path
-        csv file where the position data in StatsPerform format is saved.
+    filepath_csv: str or pathlib.Path
+        CSV file where the position data in StatsPerform format is saved.
 
     Returns
     -------
-    links: Dict[str, Dict[int, int]]
-        A link dictionary of the form ``links[team][jID] = xID``.
+    teamsheets: Dict[str, Teamsheet]
+        Dictionary with teamsheets for the home team and the away team.
     """
     # read dat-file into pd.DataFrame
-    dat_df = pd.read_csv(str(filepath_tracking))
+    csv_df = pd.read_csv(str(filepath_csv))
 
     # initialize team and ball ids
     team_ids = {"Home": 1.0, "Away": 2.0}
     ball_id = 4
 
     # check for additional tIDs
-    for ID in dat_df["team_id"].unique():
-        if not (ID in team_ids.values() or ID == ball_id):
-            warnings.warn(f"Team ID {ID} did not match any of the standard IDs!")
+    for tID in csv_df["team_id"].unique():
+        if not (tID in team_ids.values() or tID == ball_id or np.isnan(tID)):
+            warnings.warn(f"tID {tID} did not match any of the standard tIDs "
+                          f"({team_ids.values}) or the ball ID ({ball_id})!")
+
+    # initialize teamsheets
+    teamsheets = {
+        "Home": pd.DataFrame(columns=["player", "jID", "pID", "tID"]),
+        "Away": pd.DataFrame(columns=["player", "jID", "pID", "tID"]),
+    }
 
-    return _create_links_from_open_csv_df(dat_df, team_ids)
+    # loop over teams
+    for team in team_ids:
+        # extract list with pID and jID information for all players in the team
+        team_id = team_ids[team]
+        team_df = csv_df[csv_df["team_id"] == team_id]
+        jIDs = team_df["jersey_no"].unique()
+        pIDs = [
+            team_df[team_df["jersey_no"] == jID]["player_id"].unique() for jID in jIDs
+        ]
+        # possible check for multiple pIDs assgined to a single jID
+
+        # insert data to teamsheet
+        teamsheets[team]["player"] = [
+            pID[0] for pID in pIDs
+        ]
+        teamsheets[team]["jID"] = [
+            jID for jID in jIDs
+        ]
+        teamsheets[team]["pID"] = [
+            pID[0] for pID in pIDs
+        ]
```

Comment:
is this purposely the same information as for the column "player" (two above)?

Reply:
Let me know what you prefer!

Reply:
I've done the latter for the Tracab teamsheet parser (that also doesn't have any player names etc. in them), but with a capital P in "Player" and indexing starting at 1, i.e.:

``player = [f"Player {i+1}" for i in range(len(jIDs))]``


I've also added the following note to the docstring:
```
    Notes
    -----
    Tracab data does not contain any player information except jersey numbers by
    default. Thus, the teamsheet objects generated by this method will name players
    'Player i' with i starting at 1. To identify players, use the jersey numbers of
    provide custom teamsheets generated by a different parser if Tracab data is used in
    combination with other data providers.
```

Reply:
Ok, I have included this structure for both parsers. But I believe there is are two words missing in the Notes docstring, "To identify players, use the jersey numbers of [players or] provide custom teamsheets..."